src/tpm.rs

```diff
@@ -242,6 +250,66 @@ pub(crate) fn activate_credential(
     .map_err(KeylimeError::from)
 }
 
+// Returns TSS struct corresponding to an algorithm specified as a string, ex.
+// the string from the keylime.conf file.
+pub(crate) fn get_hash_alg(alg: String) -> Result<HashingAlgorithm> {
+    match alg.as_str() {
+        "Sha256" => Ok(HashingAlgorithm::Sha256),
+        other => {
+            Err(KeylimeError::Other(format!("{:?} not implemented", alg)))
+        }
+    }
+}
+
+#[derive(Debug)]
+pub(crate) enum TpmSigScheme {
+    AlgNull,
+}
+
+impl Default for TpmSigScheme {
+    fn default() -> Self {
+        TpmSigScheme::AlgNull
+    }
+}
+
+// Returns TSS struct corresponding to a signature scheme.
+pub(crate) fn get_sig_scheme(
+    scheme: TpmSigScheme,
+) -> Result<TPMT_SIG_SCHEME> {
+    match scheme {
+        // The TPM2_ALG_NULL sig scheme can be filled out with placeholder data
+        // in the details field.
+        TpmSigScheme::AlgNull => Ok(TPMT_SIG_SCHEME {
+            scheme: TPM2_ALG_NULL,
+            details: TPMU_SIG_SCHEME {
+                any: TPMS_SCHEME_HASH {
+                    hashAlg: TPM2_ALG_NULL,
+                },
+            },
+        }),
+        _ => Err(KeylimeError::Other(format!(
+            "The signature scheme {:?} is not implemented",
+            scheme
+        ))),
+    }
+}
+
+// Takes a public PKey and returns a DigestValue of it.
+pub(crate) fn pubkey_to_tpm_digest(
+    pubkey: &PKeyRef<Public>,
+    algo: HashingAlgorithm,
+) -> Result<DigestValues> {
+    match pubkey.id() {
+        Id::RSA => {
+            let mut keydigest = DigestValues::new();
+            keydigest
+                .set(algo, pubkey.rsa()?.public_key_to_pem()?.try_into()?);
+            Ok(keydigest)
+        }
+        _ => unimplemented!(),
```

Comment:
Same as other PR review, this will panic so we might want some error handling for this

Reply:
Thanks, I've changed it to an error.